src/Stratis.Bitcoin/Consensus/ChainedHeaderTree.cs

```diff
@@ -12,9 +12,23 @@ namespace Stratis.Bitcoin.Consensus
     /// <summary>
     /// TODO comment this interface
     /// </summary>
-    public interface IChainedHeaderValidator
+    public interface IChainedHeaderTreeValidator
     {
-        void Validate(ChainedHeader chainedHeader);
+        /// <summary>
+        /// Validation of a header that was seen for the first time.
+        /// </summary>
+        /// <param name="chainedHeader">The chained header to be validated.</param>
+        void HeaderValidation(ChainedHeader chainedHeader);
+
+        /// <summary>
+        /// Validate the block signature (for POS) & merkle tree, throws if validation fails. 
+        /// </summary>
+        /// <remarks>
+        /// Throws different exceptions in case any of them failed.
+        /// </remarks>
+        /// <param name="block">The block that is going to be validated.</param>
+        /// <param name="chainedHeader">The chained header of the block that will be validated.</param>
```

Comment:
No xml doc for exceptions that can be thrown

Reply:
I don't know them yet ;)